### PR TITLE
Convert <ListItemSwipeAction /> to hooks

### LINKED ISSUE #152: Convert <ListItemSwipeAction /> to hooks

### PR DESCRIPTION
**Pull request by YPAzevedo**
fixes #152 

https://user-images.githubusercontent.com/56167866/111082310-ccd68780-84c4-11eb-9458-83d9cdd742b9.mov

### CODE REVIEW

**Comment by YPAzevedo** [comment]
This is ready and working, but since there was someone else already working on this, i'll leave my PR in draft, but if we want/need to we can move it to ready for review.

File: src/components/shared/list-item-swipe-action/ListItemSwipeAction.tsx [hunk 1]
```diff
@@ -12,138 +12,135 @@ interface IListItemSwipeActionProps {
...
+      .get("pan")
+      .set({ direction: Hammer.DIRECTION_HORIZONTAL });
+    mcFrontDivRef.current.on("pan", onPan);
+
+    return () => {
+      mcFrontDivRef.current?.destroy();
     };
+  }, [lastPositionRef, onActionRef]);
+
+  return { frontDivRef, height, opacity, translateX } as const;
```
Comment (mpeyper) [inline src/components/shared/list-item-swipe-action/ListItemSwipeAction.tsx:115]: This whole hook is very similar to [what I did for `SwipeableModal`](https://github.com/OpenSourceRaidGuild/atom-pwa/blob/next/src/components/shared/swipeable-modal/SwipeableModal.tsx#L12-L117).  I wonder if we can pull out a common hook and use it in both?

Comment (YPAzevedo) [reply src/components/shared/list-item-swipe-action/ListItemSwipeAction.tsx:115]: I was thinking th same thing for the questions logic on the tests, we can see if thats possible in following prs? :)

File: src/components/shared/list-item-swipe-action/ListItemSwipeAction.tsx [hunk 2]
```diff
@@ -12,138 +12,135 @@ interface IListItemSwipeActionProps {
...
+  const [lastPosition, setLastPosition] = React.useState(0);
+  const [opacity, setOpacity] = React.useState(1);
+  const [translateX, setTranslateX] = React.useState("0%");
+
+  const frontDivRef = React.useRef<HTMLDivElement | null>(null);
+  const mcFrontDivRef = React.useRef<HammerManager | null>(null);
+  const frontDivAnimationRef = React.useRef<anime.AnimeInstance | null>(null);
+
+  // Those are things we use in the useEffect that
+  // change to often, we dont need to run the effect
```
Comment (Aprillion) [inline src/components/shared/list-item-swipe-action/ListItemSwipeAction.tsx:26]:
```suggestion
  // change too often, we don't need to run the effect
```

File: src/components/shared/list-item-swipe-action/ListItemSwipeAction.tsx (hunk 1 above)
Comment (Aprillion) [reply src/components/shared/list-item-swipe-action/ListItemSwipeAction.tsx:115]: agree it might be nice, but out of scope for  this PR which is already 🤯

**Review by Aprillion — APPROVED** [review]

**Pull request merged**